src/bots/jitMaker.ts

```diff
@@ -280,447 +168,413 @@ export class JitMakerBot implements Bot {
+							skip = true;
+						}
+						if (skip) {
+							console.log('skipping user:', userKey);
+						}
 
-		// jitter can't fill its own orders
-		if (node.userAccount.equals(userAccountPublicKey)) {
-			return false;
-		}
+						return skip;
+					});
 
-		const orderSignature = getOrderSignature(
-			node.order.orderId,
-			node.userAccount
-		);
-		const lastBaseAmountFilledSeen =
-			this.orderLastSeenBaseAmount.get(orderSignature);
-		if (lastBaseAmountFilledSeen?.eq(node.order.baseAssetAmountFilled)) {
-			return false;
-		}
+					const bestDriftBid = getBestLimitBidExcludePubKey(
+						this.dlobSubscriber.dlob,
+						perpMarketAccount.marketIndex,
+						MarketType.PERP,
+						oraclePriceData.slot.toNumber(),
+						oraclePriceData,
+						driftUser.userAccountPublicKey
+					);
 
-		return true;
-	}
+					const bestDriftAsk = getBestLimitAskExcludePubKey(
+						this.dlobSubscriber.dlob,
+						perpMarketAccount.marketIndex,
+						MarketType.PERP,
+						oraclePriceData.slot.toNumber(),
+						oraclePriceData,
+						driftUser.userAccountPublicKey
+					);
 
-	/**
-	 *
-	 */
-	private determineJitAuctionBaseFillAmount(
-		orderBaseAmountAvailable: BN,
-		orderPrice: BN
-	): BN {
-		const priceNumber = convertToNumber(orderPrice, PRICE_PRECISION);
-		const worstCaseQuoteSpend = orderBaseAmountAvailable
-			.mul(orderPrice)
-			.div(BASE_PRECISION.mul(PRICE_PRECISION))
-			.mul(QUOTE_PRECISION);
-
-		const minOrderQuote = 20;
-		let orderQuote = minOrderQuote;
-		let maxOrderQuote = convertToNumber(worstCaseQuoteSpend, QUOTE_PRECISION);
-
-		if (maxOrderQuote > this.MAX_TRADE_SIZE_QUOTE) {
-			maxOrderQuote = this.MAX_TRADE_SIZE_QUOTE;
-		}
+					// const bestDriftBid = this.dlob.estimateFillWithExactBaseAmount(
+					// 	{marketIndex: perpMarketAccount.marketIndex,
+					// 	marketType: MarketType.SPOT,
+					// 	baseAmount: baseDepth,
+					// 	orderDirection: PositionDirection.SHORT,
+					// 	slot: oraclePriceData.slot.toNumber(),
+					// 	oraclePriceData
+					// 	}
+					// ).mul(BASE_PRECISION).div(baseDepth);
+
+					// const bestDriftAsk = this.dlob.estimateFillWithExactBaseAmount(
+					// 	{marketIndex: perpMarketAccount.marketIndex,
+					// 	marketType: MarketType.PERP,
+					// 	baseAmount: baseDepth,
+					// 	orderDirection: PositionDirection.LONG,
+					// 	slot: oraclePriceData.slot.toNumber(),
+					// 	oraclePriceData
+					// 	}
+					// ).mul(BASE_PRECISION).div(baseDepth);
+
+					const bidOffset = bestDriftBid
+						.getPrice(oraclePriceData, this.dlobSubscriber.slotSource.getSlot())
+						.sub(oraclePriceData.price);
+
+					const askOffset = bestDriftAsk
+						.getPrice(oraclePriceData, this.dlobSubscriber.slotSource.getSlot())
+						.sub(oraclePriceData.price);
+
+					console.log(
+						`${convertToNumber(bidOffset, PRICE_PRECISION)}@${convertToNumber(
+							askOffset,
+							PRICE_PRECISION
+						)}`
+					);
+					if (bidOffset.gt(askOffset)) {
+						console.log('AHHHHHHHHHHHHHH');
```

Comment:
is this bad?

Reply:
whoops lol forgot to remove the console logging